vignettes/articles/glitter_for_Wikidata.Rmd

```diff
@@ -18,178 +18,225 @@ knitr::opts_chunk$set(
+  ind=which(adjacencies$adjacent==direction)
+  i=N
+  num[ind]=i
+  while(i>1){
+    indnew=which(adjacencies$adjacent==adjacencies$station[ind])
+    ind=indnew
+    i=i-1
+    num[ind]=i
+  }
+  adjacencies=adjacencies %>% 
+    mutate(num=num) %>%
+    arrange(num) 
+  adjacencies=c(adjacencies$station,
+                direction)
+  return(adjacencies)
+}
 ```
 
-We could start exploring Wikidata with this query which finds all items that are instances ("wdt:P31") of "city" or of any subclass ("wdt:P279") of "city" . This query might return many items so that it seems reasonable to limit the number of items retrieved for now with the argument `limit`
+Now let's **apply this function to all lines and directions possible**:
 
-```{r}
-spq_init() %>% 
-  spq_add("?city wdt:P31/wdt:P279* wd:Q515", .label="?city") %>%
-  spq_head(n = 10) %>% 
-  spq_perform()
+```{r calc_lines}
+stations_lines=stations_adjacency %>% 
+  sf::st_drop_geometry() %>% # make this a regular tibble, not sf
+  group_by(direction,line) %>% 
+  na.omit() %>% 
+  tidyr::nest(.key="adj") %>% # have nested "adj" table for each direction-line
+  mutate(station=purrr::map(.x=adj,.y=direction,
+                            ~form_line(.x,.y))) %>% 
+  tidyr::unnest(cols="station") %>% 
+  ungroup() %>% 
+  left_join(unique(stations_coords), # get corresponding coordinates
+            by=c("station")) %>%
+  na.omit()
+head(stations_lines)
 ```
 
-Now, let's get the location ("wdt:P625") of the cities
+`stations_lines` is now an **sf points object** which is properly formatted to be transformed into an **sf lines object** (the stations are in the right order for each line-direction, and the associated coordinates are provided in the table):
 
-```{r}
-spq_init() %>% 
-  spq_add("?city wdt:P31/wdt:P279* wd:Q515", .label="?city") %>% 
-  spq_mutate(coords = wdt::P625(city)) %>%
-  spq_head(n = 10) %>% 
-  spq_perform()
+```{r stations_lines_sf}
+stations_lines_sf=stations_lines %>% 
+  sf::st_as_sf(wkt="coords") %>% 
+  group_by(direction,line) %>% 
+  summarise(do_union=FALSE) %>%   # for each group, and keeping order of points,
+  sf::st_cast("LINESTRING")       # form a linestring geometry
+stations_lines_sf
 ```
 
-We can refine this query, stating that we want cities (or items of subclasses of city) in a radius of 5km around Lyon (which has lat-long coordinates ~ 45.76 and 4.84). We will use the argument `within_distance`:
+We can now use this new object to **display the Lyon metro lines on a leaflet map**:
 
-
-```{r}
-cities_around_Lyon = spq_init() %>% 
-  spq_add("?city wdt:P31/wdt:P279* wd:Q486972") %>% 
-  spq_label(city) %>%
-  spq_mutate(coords = wdt::P625(city),
-          .within_distance=list(center=c(long=4.84,lat=45.76),
-                               radius=5)) %>% 
-  spq_perform()
-head(cities_around_Lyon)
+```{r leaflet_lines}
+factpal <- colorFactor(topo.colors(8),
+                       unique(stations_lines$direction))
+leaflet(data=stations_sf) %>%
```

Comment:
would it be tricky to add the line number on hover?

Reply:
Maybe on hover but not as a popup. By the way I realized that having "direction" as popup rather than line was weird, then that having direction as color rather than line too. 
Here would be my suggestion:
```
factpal <- colorFactor(topo.colors(8),
                       unique(stations_lines$line))
leaflet(data=stations_sf) %>%
  addTiles() %>%
  addCircleMarkers(popup=~station, radius=3, color="red") %>% 
  addPolylines(data=stations_lines_sf,
               color=~factpal(line), popup=~line)
```